examples/demo-rollup/src/test_rpc.rs

```diff
@@ -83,44 +55,102 @@ fn test_helper(data: String, expected: &str) {
     rt.block_on(async {
         let (tx_start, rx_start) = oneshot::channel();
         let (tx_end, rx_end) = oneshot::channel();
-        let (port_sender, port_receiver) = oneshot::channel();
 
         let address = SocketAddr::new("127.0.0.1".parse().unwrap(), 0);
 
         // Initialize the ledger database, which stores blocks, transactions, events, etc.
         let tmpdir = tempfile::tempdir().unwrap();
         let mut ledger_db = LedgerDB::with_path(tmpdir.path()).unwrap();
 
-        populate_ledger(&mut ledger_db);
+        populate_ledger(&mut ledger_db, slots);
 
-        let ledger_rpc_module = ledger_rpc::get_ledger_rpc::<i32, i32>(ledger_db.clone());
+        let ledger_rpc_module = ledger_rpc::get_ledger_rpc::<u32, u32>(ledger_db.clone());
 
         rt.spawn(async move {
             let server = jsonrpsee::server::ServerBuilder::default()
                 .build([address].as_ref())
                 .await
                 .unwrap();
             let actual_address = server.local_addr().unwrap();
-            port_sender.send(actual_address.port()).unwrap();
             let _server_handle = server.start(ledger_rpc_module).unwrap();
-            tx_start.send("server started").unwrap();
+            tx_start.send(actual_address.port()).unwrap();
             rx_end.await.unwrap();
         });
 
-        rx_start.await.unwrap();
-
-        let bind_port = port_receiver.await.unwrap();
+        let bind_port = rx_start.await.unwrap();
         let rpc_config = RpcConfig {
             bind_host: "127.0.0.1".to_string(),
             bind_port,
         };
 
-        query_test_helper(data, expected, rpc_config).await;
+        query_test_helper(test_queries, rpc_config).await;
+
+        // By closing the `TempDir` explicitly we can check that it has
+        // been deleted successfully. If we don't close it explicitly,
+        // the directory will still be deleted when `tmp_dir` goes out
+        // of scope, but we won't know whether deleting the directory
+        // succeeded.
+        tmpdir.close().unwrap();
```

Comment:
But isn't that the whole purpose of the `tempdir` package? 

Also, even if deleting directory fails, should it fail our test?

Reply:
Let's remove this